src/portable/espressif/esp32s2/dcd_esp32s2.c

```diff
@@ -191,6 +212,40 @@ void dcd_init(uint8_t rhport)
                  USB_ENUMDONEMSK_M |
                  USB_RESETDETMSK_M |
                  USB_DISCONNINTMSK_M; // host most only
+  if (did_persist) {
+    USB0.grstctl &= ~USB_TXFNUM_M;
+    USB0.grstctl |= 0x10 << USB_TXFNUM_S;
+    USB0.grstctl |= USB_TXFFLSH;
+    USB0.grxfsiz = 52;
+
+    int cnt = 0;
+    do {
+      if (++cnt > 1000) {
+        ESP_LOGW(TAG, "TX FIFO flush HANG!");
+        break;
+      }
+      ets_delay_us(1);
+    } while (USB0.grstctl & USB_TXFFLSH);
+
+    for (int n = 0; n < USB_IN_EP_NUM; n++) {
+      USB0.in_ep_reg[n].diepint = USB_D_XFERCOMPL0_M | USB_D_TXFEMP0_M;
+      USB0.in_ep_reg[n].diepctl &= ~USB_D_STALL0_M; // clear Stall
+      USB0.in_ep_reg[n].diepctl |= USB_D_CNAK0 | USB_D_EPENA0; // clear NAK
+    }
+    USB0.dtknqr4_fifoemptymsk &= ~(0x7F);
+
+    USB0.gnptxfsiz = (16 << USB_NPTXFDEP_S) | (USB0.grxfsiz & 0x0000ffffUL);
+
+    USB0.daintmsk |= USB_OUTEPMSK0_M | USB_INEPMSK0_M;
+    USB0.doepmsk |= USB_SETUP0 | USB_XFERCOMPLMSK;
+    USB0.diepmsk |= USB_TIMEOUTMSK_M | USB_DI_XFERCOMPLMSK_M;//USB_INEPNAKEFFMSK
+
+    USB0.gintmsk |= USB_IEPINTMSK_M | USB_OEPINTMSK_M;
+    USB0.gotgint = ~0; //clear OTG ints
+    USB0.gintsts = ~0; //clear pending ints
+    enum_done_processing();
+    usbd_force_reconfig(rhport, 1);
```

Comment:
DCD is lowest level, it is not its responsibility to tell usbd to go into configured mode.

Reply:
yes, I accept ideas :) You know your software better. Given the position of the call, do you have a better idea of where to place it?

Reply:
maybe I can call it in our code after `tusb_init()`. Will that be OK?